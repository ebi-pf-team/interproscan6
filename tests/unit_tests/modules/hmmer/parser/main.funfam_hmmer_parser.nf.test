nextflow_process {

    name "Test Process FUNFAM_HMMER_PARSER"
    script "interproscan/modules/hmmer/parser/main.nf"
    process "FUNFAM_HMMER_PARSER"

    stage {
        symlink "tests/unit_tests/test_inputs/hmmer3_parser/funfam_4.10.75.10.out"
        symlink "interproscan/scripts/hmmer/parser_out.py"
    }

    test("Should run without failures") {

        when {
            params {}
            process {
                """
                input[0] = file("meta/tests/unit_tests/test_inputs/hmmer3_parser/funfam_4.10.75.10.out")
                input[1] = "funfam"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
