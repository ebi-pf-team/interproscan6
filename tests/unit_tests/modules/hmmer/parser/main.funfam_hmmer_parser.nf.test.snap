{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "hmmer_parsed_funfam_4.10.75.10.json:md5,803498d0da31b397194241c7a59935b7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-09-23T09:15:17.790329"
    }
}